### DISCUSSION TITLE
How to use a schedule generated from auto_schedule

### DISCUSSION THREAD

**Discussion by chkone** [halide/Halide#5405] [Q&A]
A toy example of for a Halide::Generator:
```
    void generate()
    {
        y( x ) = Halide::abs( a( x ) );
    }
```
Generate a scheduling like:
```
inline void apply_schedule_Arithmetic_Abs_f32_65536_CPU(
    ::Halide::Pipeline pipeline,
    ::Halide::Target target
) {
    using ::Halide::Func;
    using ::Halide::MemoryType;
    using ::Halide::RVar;
    using ::Halide::TailStrategy;
    using ::Halide::Var;
    Func y = pipeline.get_func(1);
    Var x(y.get_schedule().dims()[0].var);
    Var xi("xi");
    Var xii("xii");
    y
        .split(x, x, xi, 2048, TailStrategy::ShiftInwards)
        .split(xi, xi, xii, 8, TailStrategy::ShiftInwards)
        .vectorize(xii)
        .compute_root()
        .reorder({xii, xi, x})
        .parallel(x);
}
```

I wasn't able to find a example which allow me to execute this for a give buffer x ; how can I execute this generated scheduler?

**Comment by abadams** [comment]
One approach would be to include the generated header file in your generator source, and call that function like so at the end of your generate() method.

```
apply_schedule_Arithmetic_Abs_f32_65536_CPU(get_pipeline(), get_target());
```

**Comment by chkone** [comment]
I see how I can get a target for example Halide::get_target_from_environment() (or variation), but my question is how can I get a pipeline? Is it generated? I do my autoschedule with -e static_library,h,schedule.

**Comment by abadams** [comment]
get_pipeline() and get_target() are methods on Generator. If you paste the code I wrote at the end of your generate() and include the generated header with your generator source, it should work. Note that this requires recompiling the generator that was used to make the schedule in the first place.

**Comment by chkone** [comment]
I see, last question about this (I hope).
How can I send the input & get the output (I wasn't able to find examples in the code)m by toy example is just a basic dot product:
```
	constexpr int sz = 128;
	std::vector< f32 > oA( sz, 0.0f );
	std::vector< f32 > oB( sz, 0.0f );
	std::vector< f32 > oY( 1, 0.0f );

	std::transform( oA.begin(), oA.end(), oA.begin(), []( float const ){ return ( float( rand()%128 ) - 64.0f )/64.0f; } );
	std::transform( oB.begin(), oB.end(), oB.begin(), []( float const ){ return ( float( rand()%128 ) - 64.0f )/64.0f; } );

	halide_buffer_t input;
	halide_buffer_t out;

	halide_dimension_t shape[] = { {0, sz, 1} };
	halide_dimension_t shape0[] = { {0, 0, 0} };

	Halide::Buffer< f32 > bufA( &oA[ 0 ], 1, &shape[ 0 ], "a" );
	Halide::Buffer< f32 > bufB( &oB[ 0 ], 1, &shape[ 0 ], "b" );
	Halide::Buffer< f32 > bufY( &oY[ 0 ], 0, &shape0[ 0 ], "y" );
	Arithmetic_Dot_f32_0_128 oGen; // The generator

	oGen.set_inputs(bufA, bufB);
//	oGen.add_output("y")

	apply_schedule_Arithmetic_Dot_f32_128_CPU( oGen.get_pipeline(), oGen.get_target() );
```

Others attempt:
```
oGen.apply(oA, oB);
Halide::Buffer<float> im = oGen.realize(0);
```
But I keep to have the same error:
Internal Error at D:\git\vcpkg\buildtrees\halide\src\8ec94259af-565565b731.clean\src\Generator.cpp:1202 triggered by user code at : Condition failed: param_info_ptr != nullptr:
